cdc/redo/common/metric.go

```diff
@@ -23,41 +23,59 @@ const (
 )
 
 var (
-	redoWriteBytesGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+	RedoWriteBytesGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
 		Namespace: namespace,
 		Subsystem: subsystem,
 		Name:      "write_bytes_total",
 		Help:      "Total number of bytes redo log written",
 	}, []string{"namespace", "changefeed"})
 
-	redoFsyncDurationHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	RedoFsyncDurationHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Namespace: namespace,
 		Subsystem: subsystem,
 		Name:      "fsync_duration_seconds",
 		Help:      "The latency distributions of fsync called by redo writer",
 		Buckets:   prometheus.ExponentialBuckets(0.001, 2.0, 13),
 	}, []string{"namespace", "changefeed"})
 
-	redoFlushAllDurationHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	RedoFlushAllDurationHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
 		Namespace: namespace,
 		Subsystem: subsystem,
 		Name:      "flushall_duration_seconds",
 		Help:      "The latency distributions of flushall called by redo writer",
 		Buckets:   prometheus.ExponentialBuckets(0.001, 2.0, 13),
 	}, []string{"namespace", "changefeed"})
 
-	redoTotalRowsCountGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+	RedoTotalRowsCountGauge = prometheus.NewGaugeVec(prometheus.GaugeOpts{
 		Namespace: namespace,
 		Subsystem: subsystem,
 		Name:      "total_rows_count",
 		Help:      "The total count of rows that are processed by redo writer",
 	}, []string{"namespace", "changefeed"})
+
+	RedoWriteLogDurationHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: namespace,
+		Subsystem: subsystem,
+		Name:      "write_log_duration_seconds",
+		Help:      "The latency distributions of writeLog called by redoManager",
+		Buckets:   prometheus.ExponentialBuckets(0.001, 2.0, 13),
+	}, []string{"namespace", "changefeed"})
+
+	RedoFlushLogDurationHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: namespace,
+		Subsystem: subsystem,
+		Name:      "flush_log_duration_seconds",
+		Help:      "The latency distributions of flushLog called by redoManager",
+		Buckets:   prometheus.ExponentialBuckets(0.001, 2.0, 13),
+	}, []string{"namespace", "changefeed"})
 )
 
 // InitMetrics registers all metrics in this file
 func InitMetrics(registry *prometheus.Registry) {
-	registry.MustRegister(redoFsyncDurationHistogram)
-	registry.MustRegister(redoTotalRowsCountGauge)
-	registry.MustRegister(redoWriteBytesGauge)
-	registry.MustRegister(redoFlushAllDurationHistogram)
+	registry.MustRegister(RedoFsyncDurationHistogram)
+	registry.MustRegister(RedoTotalRowsCountGauge)
+	registry.MustRegister(RedoWriteBytesGauge)
+	registry.MustRegister(RedoFlushAllDurationHistogram)
+	registry.MustRegister(RedoWriteLogDurationHistogram)
+	registry.MustRegister(RedoFlushLogDurationHistogram)
```

Comment:
The two metrics `RedoFlushAllDurationHistogram` and `RedoFlushLogDurationHistogram` may be hard to differentiate.

Reply:
Compared with `RedoFlushAllDurationHistogram` , `RedoFlushLogDurationHistogram` calculated the waiting time of acquiring a lock. Maybe we could change the monitoring after we have optimized the flush logic of redo module.